src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -12427,6 +12428,30 @@ instruct cmpFastUnlock(rFlagsReg cr, rRegP object, rax_RegP box, rRegP tmp) %{
   ins_pipe(pipe_slow);
 %}
 
+instruct cmpFastLockLightweight(rFlagsReg cr, rRegP object, rbx_RegP box, rax_RegI tmp, rRegP scr) %{
+  predicate(LockingMode == LM_LIGHTWEIGHT);
+  match(Set cr (FastLock object box));
+  effect(TEMP tmp, TEMP scr, USE_KILL box);
+  ins_cost(300);
+  format %{ "fastlock $object,$box\t! kills $box,$tmp,$scr" %}
+  ins_encode %{
+    __ fast_lock_lightweight($object$$Register, $box$$Register, $tmp$$Register, $scr$$Register, r15_thread);
```

Comment:
It is slightly confusing that the names here don't match the naming in fast_lightweight_lock/unlock. You might want to fix that.

Reply:
I'll fix that. Went down a rabbit hole trying to figure out adlc and register allocation. I do not know why they specify `rbx` for `box`. Is it because they want to use `USE_KILL` or are they using `USE_KILL` because they specify `rbx` for `box`. It feels like this specification could be improved. The only requirement is that one tmp register is `rax`.  But I will leave that to another RFE.

Reply:
> I'll fix that. Went down a rabbit hole trying to figure out adlc and register allocation. I do not know why they specify `rbx` for `box`. Is it because they want to use `USE_KILL` or are they using `USE_KILL` because they specify `rbx` for `box`. It feels like this specification could be improved. The only requirement is that one tmp register is `rax`. But I will leave that to another RFE.

I don't know this, either. This might well be historical. I seem to remember that a much earlier version of that code shared code with interpreter version and used fixed registers there, but I am not sure. If it's not needed for the new code then don't do it. I don't think it has anything to do with USE_KILL, though. USE_KILL only means that it's using the input value of the register *and* overwrites (kills) it.

Reply:
Well if I remember correctly from my experiments with this `USE_KILL` requires a bound register. Will get a compilation error with the change `rbx_RegP` -> `rRegP`, saying something like `only bound registers can be killed`

Reply:
Yes, but also with TEMP?

Reply:
`TEMP` does not require a bound register. It will compile but crash later in C2.
This is only based on observed behaviour:
`TEMP` cannot be specified on an Input. Will crash in register allocation with `assert(opcnt < numopnds) failed: Accessing non-existent operand`